x-pack/plugins/observability/server/services/slo/sli_client.ts

```diff
@@ -88,6 +91,7 @@ function commonQuery(
         filter: [
           { term: { 'slo.id': slo.id } },
           { term: { 'slo.revision': slo.revision } },
+          { term: { 'slo.instanceId': instanceId } },
```

Comment:
I think we should add the filter only when the instanceId is defined and not equal to `'*'`, like we do elsewhere. E.g. !!slo.instanceId && slo.instanceId !== ALL_VALUE` then we add the filter for that particular instanceId.